### ISSUE TITLE
Continue training in a dynamic training data environment

### ISSUE THREAD

**Issue by mhnamaki** [dmlc/xgboost#1225]
Hi,
I'm working on a structured prediction problem and I want to incrementally train a model based on the wrong classified instances. Consider the case that at each state I ask the model to predict the output label. If it's wrong, then I add this instance for future learning as a new instance (dataset aggregation). So, in the next round, model will learn from these instances too (or just from these instances). 
Is there any way to do this with XGBOOST library? Specifically in Java.

Thanks.
Best.

**Comment by khotilov** [comment]
I would guess yes, but you would need to write lots of own code. And it's not quite clear whether what you want to do would conceptually align with the gradient boosting approach that xgboost implements. Also, if the "structured" part is of your main concern, xgboost's C++ code has some initial boilerplate relevant to multivariate prediction, but the functionality is not yet available.

**Comment by mhnamaki** [comment]
What about the "update" method, you've provided? Is it wrong if I change "DMatrix" in each iteration and I call update method for that?

`boosterClassifier.update(dmatSelection, round);`

At first, boosterClassifier is created like this: 
`Booster boosterClassifier = XGBoost.train(dmatSelection, paramsCls, round, watchesCls, null, null);`

**Comment by khotilov** [comment]
No, it's not wrong. There exist multiple "diversification" approaches that change data in different ways for each iterations, and you are absolutely free to try your own. The only question is how much it will help you in the end :)
